ckan/lib/helpers.py

```diff
@@ -311,7 +311,7 @@ def _get_auto_flask_context():
     from ckan.config.middleware import _internal_test_request_context
 
     # This is a normal web request, there is a request context present
-    if _request_ctx_stack.top:
+    if request_ctx:
```

Comment:
Flask has a specific function to test this. Can we use it instead?

https://flask.palletsprojects.com/en/3.0.x/api/#flask.has_request_context

Reply:
Sure, one moment